joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/frontendspecific/rubysrc2cpg/ImplicitRequirePass.scala

```diff
@@ -32,17 +45,36 @@ class ImplicitRequirePass(cpg: Cpg) extends ForkJoinParallelCpgPass[Method](cpg)
         val typeName = typeDecl.name
         typeName == fileName || typeName == CaseUtils.toCamelCase(fileName, true, '_', '-')
       }
-      .l
-    importableTypes.foreach { typeDecl =>
-      typeToPath.put(typeDecl.fullName, typeDecl.filename.replace("\\", "/").stripSuffix(".rb"))
-    }
-    importableTypes.groupBy(_.name).foreach { case (name, types) =>
-      typeNameToFullName.put(name, types.toSet)
+      .flatMap(getChildrenTypes(_, None))
+      .groupBy(_._1)
+      .foreach { case (typeName, typeImportInfos) =>
+        typeNameToImportInfo.put(typeName, typeImportInfos)
+      }
+    typeNameToImportInfo.addAll(externalTypes.map(x => (x._1, x._2, x._3, true)).groupBy(_._1))
+  }
+
+  private def typeToTypeImportInfo(typeDecl: TypeDecl, prependWith: Option[String]): Seq[TypeImportInfoInternal] = {
+    val typeName = typeDecl.name
+    if (typeName.endsWith("<class>")) {
+      Nil
+    } else {
+      // We add both "name" variants, i.e., for Foo::Bar, we have both Bar and Foo::Bar
+      (typeName +: prependWith.map(prefix => s"$prefix.${typeDecl.name}").toList).map(qualifiedTypeName =>
+        (qualifiedTypeName, typeDecl.fullName, normalizePath(typeDecl.filename), typeDecl.isExternal)
+      )
     }
   }
 
+  private def getChildrenTypes(typeDecl: TypeDecl, prependWith: Option[String] = None): Seq[TypeImportInfoInternal] = {
+    val childTypes = typeDecl.astChildren.isTypeDecl.not(_.methodBinding).l
+    typeToTypeImportInfo(typeDecl, prependWith) ++ childTypes
+      .flatMap(getChildrenTypes(_, prependWith.map(_.concat(s".${typeDecl.name}")).orElse(Option(typeDecl.name))))
```

Comment:
Incomprehensible. Please break this up in individual statements with expressive names for variables.